Packs/CommonScripts/Scripts/TextFromHTML/TextFromHTML.py

```diff
@@ -5,26 +5,57 @@
 import re
 
 
-def get_plain_text(html_regex):
+def get_plain_text(html: str):
     data = ''
-    if html_regex and html_regex.group(0):
-        data = re.sub(r'<.*?>', '', html_regex.group(0))
+    if html:
+        data = re.sub(r'<br>', '\n', html, flags=re.M + re.S + re.I + re.U)
+        data = re.sub(r'<.*?>', '', html, flags=re.M + re.S + re.I + re.U)
         entities = {'quot': '"', 'amp': '&', 'apos': "'", 'lt': '<', 'gt': '>', 'nbsp': ' ',
                     'copy': '(C)', 'reg': '(R)', 'tilde': '~', 'ldquo': '"', 'rdquo': '"', 'hellip': '...'}
         for e in entities:
             data = data.replace(f'&{e};', entities[e])
+        data = re.sub(r'[ \t]{2,}', ' ', data)
+        data = re.sub(r'(\r?\n){3,}', '\n\n', data)
+        data = re.sub(r'\s+$', '', data)
     return data
 
 
-def text_from_html(args):
-    html = args['html']
+def get_body(html: str, html_tag: str, allow_fallback: bool = False):
+    if html and html_tag:
+        body = re.search(fr'<{html_tag}.*/{html_tag}>', html, re.M + re.S + re.I + re.U)
+
+        if body and body.group(0):
+            return body.group(0)
+        elif allow_fallback and html_tag.lower() == 'body':
+            return html
+    return ''
+
+
+def text_from_html(args: dict):
+    html = args.get('html', '')
     html_tag = args.get('html_tag', 'body')
+    allow_fallback = str(args.get('allow_body_fallback', 'false')).lower() == 'true'
 
-    body = re.search(fr'<{html_tag}.*/{html_tag}>', html, re.M + re.S + re.I + re.U)
+    body = get_body(html, html_tag, allow_fallback)
     data = get_plain_text(body)
+
     return data if data != '' else 'Could not extract text'
```

Comment:
This function was kind of the `main` function before, no need to have both since it just adds unneeded complexity.
Please move this function's logic inside the new `main` function.

Reply:
I've combined them into a main function